nservicebus/messaging/delayed-delivery.md

```diff
@@ -13,6 +13,16 @@ The message doesn't have to be dispatched immediately after sending, it can be d
 
 NOTE: Only send operations can be deferred. Publish and reply operations cannot be deferred.
 
+## Handling current message later
+To handle the current message at a later time `bus.HandleCurrentMessageLater()` method can be used.  This was the initial way of *deferring* a message. This would create a copy of the message that has the same identifier, header and body, and then put the new message at the end of the queue. The message eventually will be pick up again once other messages in the queue are processed. 
+
+There are two caveats with this method:
+
+- As mentioned above a new copy of the message will be put back at the end of the queue. To make this work, the message pipeline will not abort which means any business transaction will also get comitted. 
+
+- If there are no other messages in the queue, or the condition to put the message back into the queue is not correct, the message goes back into the queue and it will be picked up immediately afterwards. This will have an effect of an endless loop which will be noted as high system resource utilization by the endpoint.
+
+As such, this method will be deprecated in NServiceBus version 7.0. It is recommended to use either [Delayed Retries](/nservicebus/recoverability/#delayed-retries) or one of the deferring mechanisms below, depending on the case at hand.  
```

Comment:
The point was to retry a message if certain condition is not met. For that purpose we can just throw a business exception, along with custom exception policies and achieve a time-bound retry instead of a endless loop.